File: servertier/src/main/scala/sgad/servertier/dataaccess/data/shareddata/Bonus.scala
/**
 * FILE: Bonus.scala
 * PERCORSO /Codice/sgad/servertier/src/main/scala/sgad/servertier/dataaccess/data/shareddata
 * DATA CREAZIONE: 18 Febbraio 2014
 * AUTORE: ProTech
 * EMAIL: protech.unipd@gmail.com
 *
 * Questo file è proprietà del gruppo ProTech, viene rilasciato sotto licenza Apache v2.
 *
 * DIARIO DELLE MODIFICHE:
 * 2014-02-18 - Creazione della classe - Biancucci Maurizio
 */
package sgad.servertier.dataaccess.data.shareddata

/**
 * Classe per la gestione dei bonus eventualmente disponibili per certi edifici.
 * @constructor
 * @param bonusName Il nome del bonus.
 * @param type Il codice identificativo del bonus, il quale determina quale caratteristica del BuildingWithLevel influenza.
 * @param quantity La potenza del bonus, cioè quanto influenza la caratteristica del BuildingWithLevel.
 */
class Bonus(private val bonusName: String, private val `type`: Int, private val quantity: Float) {
	if(bonusName==null)
		throw new IllegalArgumentException
	/**
	 * Metodo getter per l'attributo bonusName.
	 * @return Il nome del bonus.
	 */
	def getBonusName = bonusName
	/**
	 * Metodo getter per l'attributo type.
	 * @return Il codice identificativo del bonus, il quale determina quale caratteristica del BuildingWithLevel influenza.
	 */
	def getType = `type`
	/**
	 * Metodo getter per l'attributo quantity.
	 * @return  La potenza del bonus, cioè quanto influenza la caratteristica del BuildingWithLevel.
	 */
	def getQuantity = quantity
	/**
	 * Metodo per il confronto.
	 * @param other l'altro oggetto con cui fare il confronto.
	 * @return True o false in base all'esito.
	 */
	override def equals(other: Any):Boolean ={
		if(!other.isInstanceOf[Bonus])
			false
		else
			(bonusName == other.asInstanceOf[Bonus].bonusName
			&& `type` == other.asInstanceOf[Bonus].`type`
			&& quantity == other.asInstanceOf[Bonus].quantity)
	}
}

File: Codice/sgad/clienttier/model/generaldata/Bonus.js
/**
 * FILE: Bonus.js
 * PERCORSO /Codice/sgad/clienttier/model/generaldata/Bonus.js
 * DATA CREAZIONE: 12 Febbraio 2014
 * AUTORE: ProTech
 * EMAIL: protech.unipd@gmail.com
 *
 * Questo file è proprietà del gruppo ProTech, viene rilasciato sotto licenza Apache v2.
 *
 * DIARIO DELLE MODIFICHE:
 * 2014-02-13 - Implementazione dei metodi getBonusName(), getBonusQuantity(), getBonusType() e valueOf() - Gatto Francesco
 * 2014-02-12 - Creazione della classe - Gatto Francesco
 */
 
 /**
 * Classe per la gestione dei bonus eventualmente disponibili per certi edifici.
 * @constructor
 * @param {string} bonusName Rappresenta il nome del bonus.
 * @param {number} quantity Rappresenta la potenza del bonus.
 * @param {int} type Codice identificativo che rappresenta il tipo di bonus e di conseguenza i parametri che influenza con potenza pari al parametro quantity.
 **/
function Bonus(bonusName, quantity, type) {
	/**
	 * Rappresenta il nome del bonus.
	 * @type {string}
	 * @private
	 */
	this.bonusName = bonusName;
	/**
	 * Rappresenta la potenza del bonus.
	 * @type {number}
	 * @private
	 */
	this.quantity = quantity;
	/**
	 * Codice identificativo che rappresenta il tipo di bonus,
	 * e di conseguenza i parametri che influenzano con potenza pari al parametro quantity.
	 * @type {int}
	 * @private
	 */
	this.type = type;
}
/**
 * Controlla l'accesso al nome del bonus.
 * @return {string} Nome del bonus.
 */
Bonus.prototype.getBonusName = function() {
	return this.bonusName;
};
/**
 * Controlla l'accesso alla quantità del bonus.
 * @return {number} Potenza del bonus.
 */
Bonus.prototype.getBonusQuantity = function() {
	return this.quantity;
};
/**
 * Controlla l'accesso al tipo del bonus.
 * @return {int} Tipo del bonus.
 */
Bonus.prototype.getBonusType = function() {
	return this.type;
};
/**
 * Ridefinizione del metodo valueOf di Bonus
 * @override
 * @return {string} Stringa di confronto.
 */
Bonus.prototype.valueOf = function() {
    return JSON.stringify({bonusName : this.getBonusName(), quantity : this.getBonusQuantity(), type : this.getBonusType()});
};